### DISCUSSION TITLE
Use emojis to accept or reject a duel

### DISCUSSION THREAD

**Discussion by helmasaur** [helmasaur/horsengel-roulette#46] [Ideas]
For example, the emojis ✅ and ❌ could be used.

The message created by the bot after the command `!hr` used, both emojis should be added. The first one which is clicked by the provoked member should trigger the game or cancel it.

Should this change be made?

**Comment by helmasaur** [comment]
The permission `ADD_REACTIONS` should be checked (#2)

**Comment by helmasaur** [comment]
If the change is validated, an emoji has to be found for the shooting. Two kind of possibilities:

- 🔫
- 🎠, 🏇 or any other horse emoji

Something else important could be the possibility to choose between the emoji or the message trigger.

**Comment by Horsengel** [comment]
Using emojis to reply to the bot for whether or not the duel is accepted can be a good idea, however I guess it could take longer for the bot to reply. If the added delay is not a problem, then I would approve the idea.
For the emojis used, I think using the pistol emoji would be better or maybe ⚔️ , and 💨  for refusing the duel, as the person flees away from the provoker.
Finally, is it possible to have both ways of replying to the bot, so that anyone can choose how it would answer?

**Reply by helmasaur** [reply]
The time for the bot to react is set by a variable in secondes. It is me who chooses it. In the next update, there will be a random effect to make it look more realistic : for example the bot would take between 1 and 3 seconds to answer.

The bot emojis you proposed are good. I think I prefer yours actually.

I was thinking of bot way for the bot owner to decide: either all the duels are with emojis, either they are by text. We can make a test to see if it's intuitive to have both possibilities within a duel.

Now that you think it is a good idea, I can mention an issue: if the game is played with emojis, how to make the game easy to play? In fact, with emojis, I guess there would be one "card" and the players would have to interract with it. The problem is there are messages which are sand, the "card" might get lost the messages. Otherwise, we could keep a "card" per actions with an emoji reaction insteas of the textual answer. I hope I have been clear enough!

**Reply by Horsengel** [reply]
I didn't think of a possible random delay for the bot to answer, that could be a good idea, I don't seem to fully imagine it but I guess it can be cool.

I originally wasn't sure about the sword emoji, however I thought of the wind appearing in cartoons when a character flees

Yes, that can be a good idea, in fact having a test done can help us seeing possible mistakes along with having a confirmation about whether or not it is a good idea

Actually, I thought about a "card" per action, but maybe a shorter one, so that we still can see the upper messages.

Also I just thought about a crazy idea : how about a Horsengel Roulette Tournament, including more than two people and working as duels to see which one is the best. I don't know if it's a good idea

**Reply by helmasaur** [reply]
> I originally wasn't sure about the sword emoji, however I thought of the wind appearing in cartoons when a character flees

I believe it is the perfect emoji to represent a duel. I did not think about that and focused on "horse" and "revolver".

> Yes, that can be a good idea, in fact having a test done can help us seeing possible mistakes along with having a confirmation about whether or not it is a good idea
>
> Actually, I thought about a "card" per action, but maybe a shorter one, so that we still can see the upper messages.

I can make a demo version to get to see how it would look and what could be the problematics (such as the gameplay). To you,, changing from textual answers to visual answers (emoji) would be a nice upgrade?

> Also I just thought about a crazy idea : how about a Horsengel Roulette Tournament, including more than two people and working as duels to see which one is the best. I don't know if it's a good idea

This is planned (#4) and there are some problematics with such an update. For example, if there is one bullet in the revolver, there would be a winner, but who would be the loser? If you have ideas, do not hesitate to [open a discussion](https://github.com/helmasaur/horsengel-roulette/discussions/new) and to talk about multiplayer (>2), I invite you to comment the issue I linked at the beginning of the paragraph.

PS: your English is good.

**Comment by helmasaur** [comment]
New alternative: slash commands.
